client/client.go

```diff
@@ -261,6 +265,27 @@ func (c *Client) setProxy(u *url.URL, d *websocket.Dialer) error {
 	// CONNECT proxy
 	if !strings.HasPrefix(u.Scheme, "socks") {
 		d.Proxy = func(*http.Request) (*url.URL, error) {
+			if u.User != nil {
+				pass, _ := u.User.Password()
+				if pass == "" {
+					fmt.Print("*** Enter proxy password: ")
+					if term.IsTerminal(syscall.Stdin) {
```

Comment:
Does terminal case different from the read line case?

Reply:
TBH I'm not sure what the difference is (not a GO programmer...).
But when I tried this with just terminal (i.e. just the "if" part w/o else), and I tried to pass the proxy password using pipe:
`echo some_password | chisel client ....`
I got this error:
`client: Connection error: inappropriate ioctl for device`
When I googled the issue I found this thread, and the suggested solution worked: https://github.com/golang/go/issues/19909#issuecomment-461311448
See also here:
https://github.com/golang/go/issues/19909#issuecomment-400979666